src/particles.zig

```diff
@@ -0,0 +1,413 @@
+const std = @import("std");
+const main = @import("main");
+const chunk_meshing = @import("renderer/chunk_meshing.zig");
+const graphics = @import("graphics.zig");
+const SSBO = graphics.SSBO;
+const TextureArray = graphics.TextureArray;
+const Shader = graphics.Shader;
+const Image = graphics.Image;
+const game = @import("game.zig");
+const ZonElement = @import("zon.zig").ZonElement;
+const c = graphics.c;
+const random = @import("random.zig");
+const vec = @import("vec.zig");
+const Mat4f = vec.Mat4f;
+const Vec3d = vec.Vec3d;
+const Vec4d = vec.Vec4d;
+const Vec3f = vec.Vec3f;
+const Vec4f = vec.Vec4f;
+const Vec3i = vec.Vec3i;
+
+var seed: u64 = undefined;
+
+var arena = main.heap.NeverFailingArenaAllocator.init(main.globalAllocator);
+const arenaAllocator = arena.allocator();
+
+pub const ParticleManager = struct {
+	var particleTypesSSBO: SSBO = undefined;
+	var types: main.List(ParticleType) = undefined;
+	var textures: main.List(Image) = undefined;
+	var emissionTextures: main.List(Image) = undefined;
+
+	var textureArray: TextureArray = undefined;
+	var emissionTextureArray: TextureArray = undefined;
+
+	const ParticleIndex = u16;
+	var particleTypeHashmap: std.StringHashMapUnmanaged(ParticleIndex) = undefined;
```

Comment:
```suggestion
	var IdToIndex: std.StringHashMapUnmanaged(ParticleIndex) = undefined;
```

Names of those hash maps are insanely inconsistent, can we make a final decision here and apply it everywhere @IntegratedQuantum? And reverseIndeces is not the name.

Reply:
For now it's fine in my opinion. Ultimate goal would be to make a separate datastructure for tracking this anyways.

I also think idToIndex is a bit too generic in my opinion.

Reply:
> I also think idToIndex is a bit too generic in my opinion.

It is within context of ParticleManager **and** part of its internal implementation. It has solid, unambiguous context. It should always be the case. It is as generic as `textures`. I would argue opposite, `particleTypeHashmap` is overspecified, there is no other type hash map, not even any other hash map.

Reply:
> I would argue opposite, particleTypeHashmap is overspecified, there is no other type hash map, not even any other hash map.

There is no other type hash map, but there very well could be a hashmap for emitter types or emitter shapes. If you don't know the details then you wouldn't know what this hashmap is for.

Reply:
This is preparing for the future, future is unknown we can't prepare for. Also this can be solved with proper namespacing.

Reply:
Well, mainly this is just personal preference, I prefer the current way of naming things.

And like I said, we should turn this into a generic data structure and avoid the problem altogether. So let's stop discussing about it.